### PR TITLE
Fixes #2925

### LINKED ISSUE #2925: AssignedByValParameter QuickFix Rewriter leaves code as-is
Attempted to use the AssignedByValParameterMakeLocalCopyVariableQuickFix within Excel.  Essentially, the only quickfix that would modify the VBE code was the IgnoreQuickFix option.  The first two QuickFix options use the TSR and the IgnoreQuickFix still uses the ICodeModule approach.  All my unit tests pass (because the Mock VBE code fragments are successfully modified)...but in working with the real VBE, the rewriter seems unable to modify the code.  I did step through to ensure the TSR code was called and executed.

### PR DESCRIPTION
**Pull request by BZngr**
Fixes #2925.

### CODE REVIEW

File: RetailCoder.VBE/Root/RubberduckModule.cs [hunk 1]
```diff
@@ -306,6 +306,11 @@ private void BindQuickFixTypes(IEnumerable<Assembly> assemblies)
             foreach (var quickFix in quickFixes)
             {
                 Bind<IQuickFix>().To(quickFix).InSingletonScope();
+                if (quickFix.Name.Equals("AssignedByValParameterMakeLocalCopyQuickFix"))
```
Comment (Hosch250) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: Wrong place.

Comment (retailcoder) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: This is not refactoring-friendly: if/when `AssignedByValParameterMakeLocalCopyQuickFix` gets renamed, this code will break. Also, why does *that one* needs its own specific binding rule?

Comment (retailcoder) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: Wait a sec... this is a `Factory` type. There shouldn't even be a need to have a concrete implementation.

Comment (retailcoder) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: Yep. Deleting [AssignedByValParameterQuickFixDialogFactory.cs](https://github.com/rubberduck-vba/Rubberduck/blob/427ef6015b22275c178eafe10ff8d508e8c50c32/RetailCoder.VBE/UI/Refactorings/AssignedByValParameterQuickFixDialogFactory.cs) should fix this: Ninject will apply the `IFooFactory` binding convention and generate the trivial factory implementation for us.

Comment (Hosch250) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: That's probably because it follows the rules of interfaces instead of factories.

Comment (retailcoder) [inline RetailCoder.VBE/Root/RubberduckModule.cs]: @Hosch250 that would be weird. I'm not seeing how an abstract factory interface could escape this binding (regardless of which assembly it's in) if its name follows the `IFooFactory` convention:

````csharp
        // note convention: abstract factory interface names end with "Factory".
        private void ApplyAbstractFactoryConvention(IEnumerable<Assembly> assemblies)
        {
            Kernel.Bind(t => t.From(assemblies)
                .SelectAllInterfaces()
                .Where(type => type.Name.EndsWith("Factory")) 
                .BindToFactory()
                .Configure(binding => binding.InSingletonScope()));
        }
````

**Pull request closed without merging**